app/src/main/java/com/simplecity/amp_library/ui/fragments/EqualizerFragment.java

```diff
@@ -622,9 +622,7 @@ private void setupPresets() {
         eqPresetNames = new String[numPresets + 1];
 
         String[] presetNames = prefs.getString("equalizer.preset_names", "").split("\\|");
-        for (short i = 0; i < numPresets; i++) {
-            eqPresetNames[i] = presetNames[i];
-        }
+        System.arraycopy(presetNames, 0, eqPresetNames, 0, numPresets + 1);
```

Comment:
I've never trusted `arrayCopy`. That's totally irrational and unjustified - but I'd be more comfortable with this kind of change being made in a separate commit. Makes it a lot easier to revert if something goes wrong.